### PR TITLE
Update LiftedEmbedding.scala

### PR DESCRIPTION
**Pull request by mobiworx**
documenting "dynamic filter"

### CODE REVIEW

File: src/sphinx/code/LiftedEmbedding.scala [hunk 1]
```diff
@@ -190,10 +190,30 @@ object LiftedEmbedding extends App {
       //   select "COF_NAME", "SUP_ID", "PRICE", "SALES", "TOTAL"
       //     from "COFFEES"
       //     order by "COF_NAME" desc nulls first
+      
+      // building criteria using a "dynamic filter" e.g. from a webform. 
+      val criteriaColombian = Option("Colombian")
+      val criteriaEspresso = Option("Espresso")
+      val criteriaRoast:Option[String] = None
+      val criteriaDefault = LiteralColumn(1) === LiteralColumn(1)
```
Comment (cvogt) [inline src/sphinx/code/LiftedEmbedding.scala]: Couldn't you just use true here? Or true: Column[Boolean] ?

**Comment by cvogt** [comment]
Otherwise LGTM. @szeiger ?

**Comment by mobiworx** [comment]
updated to use "true:Column[Boolean]"

File: src/sphinx/code/LiftedEmbedding.scala [hunk 2]
```diff
@@ -190,10 +190,29 @@ object LiftedEmbedding extends App {
       //   select "COF_NAME", "SUP_ID", "PRICE", "SALES", "TOTAL"
       //     from "COFFEES"
       //     order by "COF_NAME" desc nulls first
+      
+      // building criteria using a "dynamic filter" e.g. from a webform. 
+      val criteriaColombian = Option("Colombian")
+      val criteriaEspresso = Option("Espresso")
+      val criteriaRoast:Option[String] = None
+
+      val q4 = coffees.map(_.name).filter { coffee =>
```
Comment (cvogt) [inline src/sphinx/code/LiftedEmbedding.scala]: Doesn't compile. I suggest you remove the `.map(_.name)`

File: src/sphinx/code/LiftedEmbedding.scala [hunk 3]
```diff
@@ -190,10 +190,29 @@ object LiftedEmbedding extends App {
...
+
+      val q4 = coffees.map(_.name).filter { coffee => 
+        List(
+            criteriaColombian.map(coffee.name === _),
+            criteriaEspresso.map(coffee.name === _),
+            criteriaRoast.map(coffee.name === _) // not a condition as `criteriaRoast` evaluates to `None` 
+        ).collect({case Some(criteria)  => criteria}).reduceLeftOption(_ || _).getOrElse(true:Column[Boolean])
+      }
+      // compiles to SQL (simplified):
+      //   select "COF_NAME"
```
Comment (cvogt) [inline src/sphinx/code/LiftedEmbedding.scala]: And change this to `select "COF_NAME", "SUP_ID", "PRICE", "SALES", "TOTAL"`

**Comment by mobiworx** [comment]
Changed it according to your proposal, works now

**Comment by cvogt** [comment]
LGTM, could you squash the commits into one, so we have a single compiling commit that adds this? Thx

**Comment by mobiworx** [comment]
Ok!

**Comment by szeiger** [comment]
Looks like a useful addition. Thanks!

**Comment by mobiworx** [comment]
What branch do I need to push this changes in order to update the docs for version 2.1?

**Comment by cvogt** [comment]
@mobiworx 3.0 is coming out imminently. Let's not distract resources from that.

**Comment by szeiger** [comment]
Since it's already in master, "2.1" would be the right branch, but at the moment there are no plans of cutting another release from that.

**Comment by mobiworx** [comment]
I do not want to trigger a new release. I just would like to provide this addition to users of 2.1. as well. Maybe it is easy possible?

**Pull request merged**